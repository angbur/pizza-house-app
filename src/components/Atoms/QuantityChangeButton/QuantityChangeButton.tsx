import { MouseEventHandler, useContext } from 'react';
import styled from 'styled-components';
import Icon from 'components/Atoms/Icon/Icon';
import { ThemeContext } from 'components/Theme/ThemeContext';

const QuantityPanel = styled.div`
  width: 24px;
  & > button {
    background: ${(props) => props.theme?.palette.light};
    color: ${(props) => props.theme?.palette.secondary};
    cursor: pointer;
    border: none;
    padding: 2px 3px;
    &: hover {
      background: ${(props) => props.theme?.palette.primary};
    }
  }
  & > button:first-child {
    border-bottom: 1px solid ${(props) => props.theme?.palette.secondary};
  }
`;

type QuantityButtonProps = {
  onIncrement?: MouseEventHandler<HTMLButtonElement>;
  onDecrement?: MouseEventHandler<HTMLButtonElement>;
};

const QuantityChangeButton = ({ onDecrement, onIncrement }: QuantityButtonProps) => {
  const theme = useContext(ThemeContext);
  return (
    <QuantityPanel theme={theme}>
      <button onClick={onIncrement}>{Icon({ name: 'arrow-up', size: 14, isActive: true })}</button>
      <button onClick={onDecrement}>
        {Icon({ name: 'arrow-down', size: 14, isActive: true })}
      </button>
    </QuantityPanel>
  );
};

export default QuantityChangeButton;
