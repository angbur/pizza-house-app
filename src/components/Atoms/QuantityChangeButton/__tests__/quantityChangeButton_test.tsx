import renderer from 'react-test-renderer';
import ThemeProvider from 'components/Theme/ThemeContext';
import QuantityChangeButton from '../QuantityChangeButton';

const Wrapper = () => {
  return (
    <ThemeProvider>
      <QuantityChangeButton onIncrement={jest.fn()} onDecrement={jest.fn()} />
    </ThemeProvider>
  );
};

it('renders correctly quantity change button', () => {
  const tree = renderer.create(<Wrapper />).toJSON();
  expect(tree).toMatchInlineSnapshot(`
    <div
      className="sc-bcXHqe eqesaf"
    >
      <button
        onClick={[MockFunction]}
      >
        <img
          height={14}
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "arrow-up-active.svg",
            }
          }
        />
      </button>
      <button
        onClick={[MockFunction]}
      >
        <img
          height={14}
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "arrow-down-active.svg",
            }
          }
        />
      </button>
    </div>
  `);
});
