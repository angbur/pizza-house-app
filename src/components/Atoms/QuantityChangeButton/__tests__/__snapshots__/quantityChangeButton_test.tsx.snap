// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly quantity change button 1`] = `
<div
  className="sc-bcXHqe eqesaf quantity-button"
>
  <button
    onClick={[MockFunction]}
  >
    <img
      height={14}
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "arrow-up-active.svg",
        }
      }
    />
  </button>
  <button
    onClick={[MockFunction]}
  >
    <img
      height={14}
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "arrow-down-active.svg",
        }
      }
    />
  </button>
</div>
`;
